ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Wallets.hs

```diff
@@ -175,3 +199,21 @@ updateHdWalletAssurance
     -> IO (Either HD.UnknownHdRoot HdRoot)
 updateHdWalletAssurance pw hdRootId assuranceLevel =
     update' (pw ^. wallets) (UpdateHdWalletAssurance hdRootId assuranceLevel)
+
+{-------------------------------------------------------------------------------
+  Automatic creation of Wallet Account and Address
+-------------------------------------------------------------------------------}
+
+addDefaultAddress 
+    :: PassiveWallet
+    -> WalletId
+    -> PassPhrase
+    -> ExceptT CreateWalletError IO ()
+addDefaultAddress pw walletId passphrase = do 
+    res <- liftIO $ createAccount (AccountName "Unnamed account") walletId pw
+    case res of 
+        Left err      -> throwError $ CreateAccountFailed err
+        Right account -> do
```

Comment:
This is better. You can also use `liftEither` like [here](https://github.com/serokell/ariadne/blob/624b379e508fcc926eca3dc8cf4c2af1d02a7147/ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Accounts.hs#L106), but that's not essential.

Reply:
I don't think we can directly use liftEither as in the example you sited,  since `createAccount` returns  an `IO` with a different Exception type. I ended up refactoring it differently all the same.